src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -18,7 +18,7 @@ namespace Raven.Server.Documents.Indexes.MapReduce.Static
 {
     public class MapReduceIndex : MapReduceIndexBase<MapReduceIndexDefinition>
     {
-        private readonly HashSet<CollectionName> _referencedCollections = new HashSet<CollectionName>();
+        private readonly HashSet<CollectionName> _referencedCollections = new HashSet<CollectionName>(CollectionNameComparer.Instance);
```

Comment:
But we are calling ```.Count```. Unless we have a better way to do that, ```HashSet<T>``` is the best I can think of.